File: JSPM.Web/Modules/ServerTypes/Students/SkillsForm.ts
import { StringEditor, IntegerEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface SkillsForm {
    SkillCategory: StringEditor;
    Rating: IntegerEditor;
    AknowledgeBy: StringEditor;
    Certification: StringEditor;
    CertificationDetails: StringEditor;
    StudentId: IntegerEditor;
}

export class SkillsForm extends PrefixedContext {
    static formKey = 'Students.Skills';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SkillsForm.init)  {
            SkillsForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;

            initFormType(SkillsForm, [
                'SkillCategory', w0,
                'Rating', w1,
                'AknowledgeBy', w0,
                'Certification', w0,
                'CertificationDetails', w0,
                'StudentId', w1
            ]);
        }
    }
}

File: JSPM.Web/Modules/Students/Skills/SkillsForm.cs
using Serenity.ComponentModel;

namespace JSPM.Students.Forms;

[FormScript("Students.Skills")]
[BasedOnRow(typeof(SkillsRow), CheckNames = true)]
public class SkillsForm
{
    public string SkillCategory { get; set; }
    public int Rating { get; set; }
    public string AknowledgeBy { get; set; }
    public string Certification { get; set; }
    public string CertificationDetails { get; set; }
    public int StudentId { get; set; }
}